duckduckgo/duckduckgo-privacy-extension
# Add upgradeToSecure support for Firefox 59+

**Reviewer:** @jdorweiler 



## Description:

Mozilla released a fix in Firefox 59 for the http -> https upgrade cors errors we've been seeing. This PR implements the fix for Firefox 59+ users. Unfortunately, there doesn't appear to be a way to fix this issue in older versions of Firefox.

## Steps to test this PR:

1. With privacy protection on, visit a site in Firefox 59+ that has been breaking due to cors error. That is, check an http site where we are upgrading child requests to https. I've been using http://www.thrashermagazine.com/ and http://www.balaodainformatica.com.br/ to test. The site should no longer be broken.
2. Repeat 1, using chrome. Site should be ok.
3. `npm run test`


## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
